dev/reference/alert-rules.md

```diff
@@ -0,0 +1,1276 @@
+
+## TiKV alert rules
+
+This section gives the alert rules for the TiKV component.
+
+### Emergency-level alerts
+
+Emergency-level alerts are often caused by a service or node failure. Manual intervention is required immediately.
+
+#### `TiKV_memory_used_too_fast`
+
+* Alert rule:
+
+    `process_resident_memory_bytes{job=~"tikv",instance=~".*"} - (process_resident_memory_bytes{job=~"tikv",instance=~".*"} offset 5m) > 5*1024*1024*1024`
+
+* Description:
+
+    Currently, there are no TiKV monitoring items that are about memory. You can monitor the memory usage of the machines in the cluster by node_exporter. The above rule indicates that when the memory usage exceeds 5 GB within 5 minutes (the memory is occupied too fast in TiKV), an alert is triggered.
+
+* Solution:
+
+    Adjust the `block-cache-size` of `rockdb.defaultcf` and `rocksdb.writecf`.
+
+#### `TiKV_GC_can_not_work`
+
+* Alert rule:
+
+    `sum(increase(tidb_tikvclient_gc_action_result{type="success"}[6h])) < 1`
+
+* Description:
+
+    GC is not performed successfully in a Region within 6 hours, which indicates that GC is not working properly. If GC does not run in a short term, it will not cause much trouble; but if GC keeps down, more and more versions are retained, which slows down the query.
+
+* Solution:
+
+    1. Perform `select VARIABLE_VALUE from mysql.tidb where VARIABLE_NAME = "tikv_gc_leader_desc"` to locate the `tidb-server` corresponding to the GC leader;
+    2. View the log of the `tidb-server`, and grep gc_worker tidb.log;
+    3. If you find that the GC worker has been resolving locks (the last log is "start resolve locks") or deleting ranges (the last log is “start delete {number} ranges”) during this time, it means the GC process is running normally. Otherwise, contact `support@pingcap.com` to resolve this issue.
+
+### Critical-level alerts
+
+For the critical-level alerts, a close watch on the abnormal metrics is required.
+
+#### `TiKV_server_report_failure_msg_total`
+
+* Alert rule:
+
+    `sum(rate(tikv_server_report_failure_msg_total{type="unreachable"}[10m])) BY (store_id) > 10`
+
+* Description:
+
+    Indicates that the remote TiKV cannot be connected.
+
+* Solution:
+
+    1. Check whether the network is clear.
+    2. Check whether the remote TiKV is down.
+    3. If the remote TiKV is not down, check whether the pressure is too high. Refer to the solution in [`TiKV_channel_full_total`](#-tikv-channel-full-total).
+
+#### `TiKV_channel_full_total`
+
+* Alert rule:
+
+    `sum(rate(tikv_channel_full_total[10m])) BY (type, instance) > 0`
+
+* Description:
+
+    This issue is often caused by the stuck Raftstore thread and high pressure on TiKV.
+
+* Solution:
+
+    1. Watch the Raft Propose monitor, and see whether the alerted TiKV node has a much higher raft propose than other TiKV nodes. If so, it means that there are one or more hot spots on this TiKV. You need to check whether the hot spot scheduling can work properly.
```

Comment:
```suggestion
    1. Watch the Raft Propose monitor, and see whether the alerted TiKV node has a much higher Raft propose than other TiKV nodes. If so, it means that there are one or more hot spots on this TiKV. You need to check whether the hot spot scheduling can work properly.
```

Reply:
`raft` should be replaced by "Raft"?

Reply:
The rest of the comments have been resolved. PTAL again @lilin90

Reply:
> "raft" should be replaced by "Raft"? @lilin90

Yes, I think so. Ref: https://raft.github.io/